rules-reviewed/eap7/eap6/hibernate4-xml.windup.xml

```diff
@@ -40,5 +45,109 @@
+                <hint title="Class org.hibernate.id.{idclass} was removed/deprecated in Hibernate 5" effort="1" severity="mandatory">
+                    <message>Use `org.hibernate.id.enhanced.SequenceStyleGenerator` instead.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#deprecations"
+                        title="Removed or deprecated types in Hibernate 5" />
+                    <link
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes#hibernate_and_jpa_migration_changes"
+                        title="Hibernate and JPA migration changes" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+            <where param="idclass">
+                <matches pattern="(TableGenerator|TableHiLoGenerator|SequenceGenerator|SequenceIdentityGenerator|SequenceHiLoGenerator)" />
+            </where>
+        </rule>
+        <rule id="hibernate4-xml-00003">
+            <when>
+                <xmlfile 
+                   matches="//generator/@class[windup:matches(self::node(), '{generator}')]"
+                 in="{*}.hbm.xml"/>
+            </when>
+            <perform>
+               <hint title="Identifier generator {generator} was removed/deprecated in Hibernate 5" effort="1" severity="mandatory">
+                  <message>Identifier generator {generator} was removed/deprecated in Hibernate 5. You can use `sequence` generator instead.</message>
+                    <link
+                        href="https://github.com/hibernate/hibernate-orm/blob/5.0/migration-guide.adoc#deprecations"
+                        title="Removed or deprecated types in Hibernate 5" />
+                    <link
+                        href="https://access.redhat.com/documentation/en/red-hat-jboss-enterprise-application-platform/7.0.beta/migration-guide/chapter-4-application-changes#hibernate_and_jpa_migration_changes"
+                        title="Hibernate and JPA migration changes" />
+                    <tag>hibernate</tag>
+                </hint>
+            </perform>
+            <where param="generator">
+                <matches pattern="(sequence|sequence-identity|hilo)" />
+            </where>
+        </rule>
+        <rule id="hibernate4-xml-00004">
+            <when>
+                <xmlfile matches="//property[text() = 'org.hibernate.hql.spi.PersistentTableBulkIdStrategy']" as="cfg" />
+                <filecontent pattern="hibernate.hql.bulk_id_strategy=org.hibernate.hql.spi.TemporaryTableBulkIdStrategy" filename="hibernate.properties" as="properties" />
+            </when>
+            <perform>
+                <iteration over="properties">
+                    <hint title="TemporaryTableBulkIdStrategy was replaced in Hibernate 5" effort="1" severity="mandatory">
```

Comment:
The rule for `PersistentTableBulkIdStrategy` has a message `Class PersistentTableBulkIdStrategy was moved in Hibernate 5` and this rule has `TemporaryTableBulkIdStrategy was replaced in Hibernate 5`. The messages should IMHO be consistent.